core/cache/package_test.go

```diff
@@ -66,6 +67,86 @@ func (s *BaseSuite) NewHub() *pubsub.SimpleHub {
 	return pubsub.NewSimpleHub(&pubsub.SimpleHubConfig{Logger: logger})
 }
 
+func (s *BaseSuite) New(c *gc.C) (*Controller, <-chan interface{}) {
+	events := s.CaptureEvents(c)
+	controller, err := s.NewController()
+	c.Assert(err, jc.ErrorIsNil)
+	s.AddCleanup(func(c *gc.C) { workertest.CleanKill(c, controller) })
+	return controller, events
+}
+
+func (s *BaseSuite) CaptureEvents(c *gc.C) <-chan interface{} {
+	events := make(chan interface{})
+	s.Config.Notify = func(change interface{}) {
+		send := false
+		switch change.(type) {
+		case ModelChange:
```

Comment:
https://stackoverflow.com/questions/47929276/implement-logical-or-for-types-within-switch-in-golang says that this should work:
```
case ModelChange, RemoveModel, ApplicationChange, RemoveApplication:
  send = true
```